xbmc/settings/lib/SettingsManager.cpp

```diff
@@ -1248,7 +1248,9 @@ void CSettingsManager::ResolveReferenceSettings(std::shared_ptr<CSettingSection>
         else
         {
           referencedSetting = itReferencedSetting->second.setting;
-          m_settings.erase(FindSetting(referenceSetting->GetId()));
+          auto itReferencedSetting2 = FindSetting(referenceSetting->GetId());
```

Comment:
Oh yes, you are right. Becomes changed.

Is this a fix enough, or maybe another part who need change?